src/Dialect/GherkinDialect.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Gherkin\Dialect;
+
+/**
+ * @phpstan-type TDialectData array{
+ *     feature: non-empty-list<non-empty-string>,
+ *     background: non-empty-list<non-empty-string>,
+ *     scenario: non-empty-list<non-empty-string>,
+ *     scenarioOutline: non-empty-list<non-empty-string>,
+ *     examples: non-empty-list<non-empty-string>,
+ *     rule: non-empty-list<non-empty-string>,
+ *     given: non-empty-list<non-empty-string>,
+ *     when: non-empty-list<non-empty-string>,
+ *     then: non-empty-list<non-empty-string>,
+ *     and: non-empty-list<non-empty-string>,
+ *     but: non-empty-list<non-empty-string>,
+ * }
+ */
+final class GherkinDialect
+{
+    /**
+     * @var non-empty-list<non-empty-string>|null
+     */
+    private ?array $stepKeywordsCache = null;
```

Comment:
I would potentially just make it `private array $stepKeywordsCache;` so that it is clearly not initialised. 

The getter can still safely do `if (isset($this->stepKeywordsCache))` or `$this->stepKeywordsCache ??=` etc to check if the cache is/needs to be initialised, but the code that populates the cache is guaranteed to assign an array, and it makes clear that the property is only expected to be accessed through the getter/initialiser method?

Reply:
Ah, I hadn't realised phpstan wouldn't warn on unsafe access to the uninitialized property. In that case I agree, `?array = null` is good.